### PR TITLE
Account creation

### PR DESCRIPTION
**Pull request by dafuga**
Adds account creation to the `tools/utilities` section.

### CODE REVIEW

File: app/shared/components/Global/Form/Field/Key/Public.js [hunk 1]
```diff
@@ -6,25 +6,37 @@ import { Form, Input } from 'semantic-ui-react';
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      value: props.defaultValue
+    };
+  }
+
   onChange = debounce((e, { name, value }) => {
     const parsed = value.trim();
+    const valid = !!(parsed.match(/^EOS[a-zA-Z1-9.]{50}?$/g));
```
Comment (aaroncox) [inline app/shared/components/Global/Form/Field/Key/Public.js]: Don't use regex here - you should use:

```
// in header
const ecc = require('eosjs-ecc');

//  in onChange
const valid = ecc.isValidPublic(pubkey);
```

File: app/shared/components/Global/Form/Field/Key/Public.js [hunk 2]
```diff
@@ -6,25 +6,37 @@ import { Form, Input } from 'semantic-ui-react';
...
+
     this.setState({
       value: parsed
     }, () => {
-      this.props.onChange(e, { name, value: parsed });
+      this.props.onChange(e, { name, value: parsed, valid });
     });
-  }, 300)
+  }, 1000)
```
Comment (aaroncox) [inline app/shared/components/Global/Form/Field/Key/Public.js]: Try reducing this back down to 300 with `the ecc.isValidPublic` call.

File: app/shared/components/Tools/CreateAccount.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+      validate,
+      wallet,
+      t
+    } = this.props;
+
+    const account = accounts[settings.account];
+
+    return (
+      <Grid centered>
+        <Grid.Column width={8} style={{ textAlign: 'center' }} >
```
Comment (aaroncox) [inline app/shared/components/Tools/CreateAccount.js]: Instead of instantly using the `style` parameter for things like this, check the documentation:

https://react.semantic-ui.com/collections/grid

This should be:

```
<Grid.Column width={8} textAlign="center">
```

File: app/shared/components/Tools/Form/CreateAccount.js [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+        globals,
+        system
+      } = this.props;
+
+      const {
+        checkAccountAvailability
+      } = actions;
+
+      if (name === 'accountName' &&
+          accountName.length === 12 &&
```
Comment (aaroncox) [inline app/shared/components/Tools/Form/CreateAccount.js]: Didn't we talk about not requiring 12 characters?

File: app/shared/components/Tools/Form/CreateAccount.js [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+        decimalDelegatedBw.lessThan(1) &&
+        decimalDelegatedBw.greaterThan(0)) ||
+      (decimalDelegatedCpu &&
+        decimalDelegatedCpu.lessThan(1) &&
+        decimalDelegatedCpu.greaterThan(0));
+
+    return (
+      <Segment
+        loading={system.CREATEACCOUNT === 'PENDING'}
+        style={{ minHeight: '100px' }}
```
Comment (aaroncox) [inline app/shared/components/Tools/Form/CreateAccount.js]: Why is this here? I don't think we need any minimum heights do we?

File: app/shared/components/Tools/Form/CreateAccount.js (hunk 4 above)
Comment (dafuga) [reply app/shared/components/Tools/Form/CreateAccount.js]: This is not requiring it, but rather displaying a warning saying that only characters that have been bid on can have less than 12 characters.

File: app/shared/components/Global/Form/Field/Account.js [hunk 6]
```diff
@@ -9,10 +9,10 @@ export default class GlobalFormFieldAccount extends Component<Props> {
   constructor(props) {
     super(props);
     this.state = {
-      value: props.value
+      value: props.defaultValue
```
[the remark this answers is not in the document]
Comment (dafuga) [reply app/shared/components/Global/Form/Field/Account.js]: I was getting this warning `warning.js:33 Warning: A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components` and the change didn't seem to break other components. I can change it back though.

File: app/shared/components/Global/Form/Field/Key/Public.js (hunk 1 above)
Comment (dafuga) [reply app/shared/components/Global/Form/Field/Key/Public.js]: cool, will do 👍

File: app/shared/components/Global/Form/Field/Key/Public.js (hunk 2 above)
Comment (dafuga) [reply app/shared/components/Global/Form/Field/Key/Public.js]: I actually meant to change that back and forgot lol, thanks for pointing it out.

File: app/shared/components/Tools/CreateAccount.js (hunk 3 above)
Comment (dafuga) [reply app/shared/components/Tools/CreateAccount.js]: thanks, didn't know about this!

File: app/shared/components/Tools/Form/CreateAccount.js (hunk 5 above)
Comment (dafuga) [reply app/shared/components/Tools/Form/CreateAccount.js]: Nice catch! Copied that over from the staking form where it use to be required before we refactored staking to use your `GlobalTransactionHandler` component. I'll remove it here and from the staking form.

File: app/shared/components/Global/Form/Field/Account.js [hunk 7]
```diff
@@ -52,7 +52,7 @@ export default class GlobalFormFieldAccount extends Component<Props> {
         name={name}
         onChange={this.onChange}
         ref={ref => { this.input = ref; }}
-        value={value}
+        defaultValue={value}
```
[the remark this answers is not in the document]
Comment (dafuga) [reply app/shared/components/Global/Form/Field/Account.js]: I was getting this warning `warning.js:33 Warning: A component is changing an uncontrolled input of type text to be controlled. Input elements should not switch from uncontrolled to controlled (or vice versa). Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://fb.me/react-controlled-components` and the change didn't seem to break other components. I can change it back though.

File: app/shared/components/Tools/Form/CreateAccount.js (hunk 4 above)
Comment (dafuga) [reply app/shared/components/Tools/Form/CreateAccount.js]: sorry this is actually only checking account availability for accounts that have 12 characters, since most users are going to use account names with 12 characters, I figured that it would be better not to the account name lookup until all 12 characters are entered. I'm down for whatever you think is best though 😄

Comment (aaroncox) [reply app/shared/components/Tools/Form/CreateAccount.js]: Yeah, I'd remove that. The minute someone tries to create a non-12 character name, the lookup isn't going to work.

File: app/shared/components/Global/Form/Field/Account.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (aaroncox) [reply app/shared/components/Global/Form/Field/Account.js]: Yeah I'd undo all the value stuff possible that was changed here. I think there's a better way of fixing that.

[the remark this answers is not in the document]
Comment (dafuga) [reply app/shared/components/Global/Form/Field/Account.js]: sounds good!

File: app/shared/components/Tools/Form/CreateAccount.js (hunk 4 above)
Comment (dafuga) [reply app/shared/components/Tools/Form/CreateAccount.js]: 👍

**Comment by dafuga** [comment]
Thanks @aaroncox !! Made all the discussed changes.

**Comment by aaroncox** [comment]
Thanks for the hard work on this! Going to merge - and we can make further changes we find through testing as additional PRs.
